### ISSUE TITLE
plot_xy gives 2 plot in a jupyter notebook

### ISSUE THREAD

**Issue by dvincentwest** [gb119/Stoner-PythonCode#9]
If I run the following code in a notebook, it plots the same thing twice, once before the out and once after

```
fname = r"C:\path_to_file\xrd_file.ras"
spc = Stoner.FileFormats.RigakuFile(fname)
p = Stoner.plot.PlotFile(spc)
p.setas("xye")
p.plot_xy()
```

**Comment by gb119** [comment]
This is happening because the plot_* commands return a matplotlib figure instance as well as actually doing the plotting. The notebook then is picking up the figure instance being returned from the plot_xy function and showing that - hence you get two figures. If you simply assign the output of plot_xy to a variable this won't happen (and you can then fiddle with the figure directly as well!).

In an ipython console the return value representation is just given in a textual form so this isn't an issue. Perhaps I should add some form of  class attribute to control this behaviour.

**Comment by gb119** [comment]
PlotMixin.showfig now controls this - set it to None for use in an Jupyter notebook to suppress the return value from the plot functions.

**Comment by gb119** [comment]
I'll add a unit test to ensure that this remains working and then close this issue.

**Comment by gb119** [comment]
Fixed in d7dbd29..1f7b1b2
